### PR TITLE
docs(Core): improve several functions documentation

### PR DESCRIPTION
**Pull request by Grimdhex**
## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

* focus mainly for dev script usage, I've improve the documentation of many common functions
* Also revamp some comments and few variable names here and there

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## Known Issues and TODO List:

- Any review, comments or help to improve this PR is welcome

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -2370,7 +2374,7 @@ SpellInfo const* Creature::reachWithSpellCure(Unit* victim)
     return nullptr;
 }
 
-// select nearest hostile unit within the given distance (regardless of threat list).
+// @brief Select nearest hostile unit within the given distance (regardless of threat list).
```
Comment (Kitzunu) [inline src/server/game/Entities/Creature/Creature.cpp:2377]: you need 3 forwardslashes for doxygen to pick it up, alternative /**

File: src/server/game/Entities/Creature/Creature.h [hunk 2]
```diff
@@ -310,8 +316,8 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
     [[nodiscard]] bool hasInvolvedQuest(uint32 quest_id)  const override;
 
     bool isRegeneratingHealth() { return m_regenHealth; }
-    void SetRegeneratingHealth(bool c) { m_regenHealth = c; }
-    void SetRegeneratingPower(bool c) { m_regenPower = c; }
+    void SetRegeneratingHealth(bool creature) { m_regenHealth = creature; }
+    void SetRegeneratingPower(bool creature) { m_regenPower = creature; }
```
Comment (Kitzunu) [inline src/server/game/Entities/Creature/Creature.h:320]: creature doesn't really make sense here as you aren't passing a creature object. `bool on` would make more sense

Comment (Kitzunu) [reply src/server/game/Entities/Creature/Creature.h:320]: or enable, or something along those lines

Comment (Grimdhex) [reply src/server/game/Entities/Creature/Creature.h:320]: I didn't pay attention to the variable type. It's fix.

**Comment by Grimdhex** [comment]
Can be review. It's not all there, but it's a good start. Other PRs will come later when I'm motivated enough.

File: src/server/game/AI/CoreAI/UnitAI.h [hunk 3]
```diff
@@ -198,14 +198,14 @@ class UnitAI
     virtual ~UnitAI() {}
 
     virtual bool CanAIAttack(Unit const* /*target*/) const { return true; }
-    virtual void AttackStart(Unit* /*target*/);
+    virtual void AttackStart(Unit* /*target*/);    /// @biref Use to start attacking a target. Called just before JustEngagedWith()
```
Comment (Nyeriah) [inline src/server/game/AI/CoreAI/UnitAI.h:201]: brief?

File: src/server/game/AI/CoreAI/UnitAI.cpp [hunk 4]
```diff
@@ -253,6 +253,9 @@ SpellCastResult UnitAI::DoCastAOE(uint32 spellId, bool triggered)
     return me->CastSpell((Unit*)nullptr, spellId, triggered);
 }
 
+/**
+ * @biref Cast the spell on a random unit from the threat list
```
Comment (Nyeriah) [inline src/server/game/AI/CoreAI/UnitAI.cpp:257]: same here

File: src/server/game/AI/CoreAI/UnitAI.h [hunk 5]
```diff
@@ -358,42 +358,44 @@ class UnitAI
...
-    // for attack reaction use AttackedBy called for not DOT damage in Unit::DealDamage also
+    /** @brief Called at any Damage from any attacker (before damage apply)
+     *
+     *  @note It use for recalculation damage or special reaction at damage
+     *  for attack reaction use AttackedBy called for non DOT damage in Unit::DealDamage also
+     */
     virtual void DamageTaken(Unit* /*attacker*/, uint32& /*damage*/, DamageEffectType /*damagetype*/, SpellSchoolMask /*damageSchoolMask*/) {}
 
-    // Called when the creature receives heal
+    /// @biref Called when the creature receives heal
```
Comment (Nyeriah) [inline src/server/game/AI/CoreAI/UnitAI.h:371]: brief

File: src/server/game/AI/CoreAI/UnitAI.h (hunk 3 above)
Comment (Grimdhex) [reply src/server/game/AI/CoreAI/UnitAI.h:201]: Doxygen usage (https://www.cs.cmu.edu/~410/doc/doxygen.html). @Brief give a short description of the method. @param describe parameters... and there are more keyword that we can use (@note, @return....). It's also use by some IDE (Visual Studio by example).

**Comment by Nyeriah** [comment]
Sorry I wasn't clear. I know what it is. I am referring to the typos.

**Comment by Grimdhex** [comment]
Fixed

File: src/server/game/Entities/Creature/Creature.cpp [hunk 6]
```diff
@@ -2386,7 +2392,9 @@ Unit* Creature::SelectNearestTarget(float dist, bool playerOnly /* = false */) c
     return target;
 }
 
-// select nearest hostile unit within the given attack distance (i.e. distance is ignored if > than ATTACK_DISTANCE), regardless of threat list.
+/**
+ * @biref Select nearest hostile unit within the given attack distance (i.e. distance is ignored if > than ATTACK_DISTANCE), regardless of threat list.
```
Comment (Nyeriah) [inline src/server/game/Entities/Creature/Creature.cpp:2396]:
```suggestion
 * @brief Select nearest hostile unit within the given attack distance (i.e. distance is ignored if > than ATTACK_DISTANCE), regardless of threat list.
```

File: src/server/game/Entities/Creature/Creature.h [hunk 7]
```diff
@@ -266,7 +272,7 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
     bool _IsTargetAcceptable(Unit const* target) const;
     [[nodiscard]] bool CanIgnoreFeignDeath() const { return (GetCreatureTemplate()->flags_extra & CREATURE_FLAG_EXTRA_IGNORE_FEIGN_DEATH) != 0; }
 
-    // pussywizard: updated at faction change, disable move in line of sight if actual faction is not hostile to anyone
+    // Updated at faction change, disable move in line of sight if actual faction is not hostile to anyone
```
Comment (Nyeriah) [inline src/server/game/Entities/Creature/Creature.h:275]: Please keep those, they are a good indicator of what came from sunwell, and what we may need to change or remove in the future

File: src/server/game/AI/CoreAI/UnitAI.h (hunk 3 above)
Comment (Nyeriah) [inline src/server/game/AI/CoreAI/UnitAI.h:201]:
```suggestion
    virtual void AttackStart(Unit* /*target*/);    /// @brief Use to start attacking a target. Called just before JustEngagedWith()
```

**Comment by Nyeriah** [comment]
There are a couple of typos still. Left comments on them. Also please don`t remove sunwell comments, they're useful to keep track of what was added by the old team, and might need to be revisited in the future if issues show up.

**Comment by Grimdhex** [comment]
sorry for the two omissions. I used ‘Search in files’ to make sure there are none left. I've restored the old names on some comments as indicated. I've left removed: "// Xinef: do not allow to move with UNIT_FLAG_DISABLE_MOVE" , as I've included them in the brief.

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**